go/private/rules/cross.bzl

```diff
@@ -0,0 +1,141 @@
+# Copyright 2014 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load(
+    "//go/private/rules:transition.bzl",
+    "go_cross_transition",
+)
+load(
+    "//go/private:providers.bzl",
+    "GoArchive",
+    "GoLibrary",
+    "GoSource",
+)
+
+def _is_windows(ctx):
+    return ctx.configuration.host_path_separator == ";"
```

Comment:
This is problematic since it checks for the OS of the host platform, not the target platform. You could instead use `ctx.target_platform_has_constraint` and implicitly depend on `@platforms//os:windows".

Thinking about it some more, I'm not sure whether `.bat` files can simply be executed on Windows without a launcher. Have you been able to test that?

Reply:
I assumed that `bazel run` would run things on the host platform and not the target platform, is that assumption wrong?

I haven't tested it manually, however, I think the `non_executable_test` shouldn't pass on the windows CI if it doesn't work.

Reply:
`bazel run` only runs on the host, but the `executable` may also be (accidentally) run when used as e.g. a tool during the build or called from a test. If it turns out to be too difficult, this is not absolutely necessary though.

Agreed, if this passed on Windows CI, it's probably fine.

Reply:
So should I just use `target_platform_has_constraint` and not worry about the case where someone is trying to `bazel run` something for a different target than the host?

Reply:
Good point. Given that this rules is all about cross-compilation, a helpful error with `bazel run` is probably more important - we can keep this as is.